src/common/post.ts

```diff
@@ -49,3 +52,29 @@ export const getPostCommenterIds = async (
 
 export const hasAuthorScout = (post: Post): boolean =>
   !!post?.authorId || !!post?.scoutId;
+
+export const postScraperOrigin = process.env.POST_SCRAPER_ORIGIN;
+
+export const fetchLinkPreview = async (
+  url: string,
+): Promise<ExternalLinkPreview> => {
+  if (!isValidHttpUrl(url)) {
+    throw new ValidationError('URL is not valid');
+  }
+
+  const res = await fetch(`${postScraperOrigin}/preview`, {
+    method: 'POST',
+    headers: { 'Content-Type': '' },
+    body: JSON.stringify({ url }),
+  });
+
+  if (res.status >= 400 && res.status < 500) {
+    throw new ValidationError('Bad request!');
+  }
+
+  if (res.status >= 500) {
+    throw new ValidationError('Internal server error!');
```

Comment:
This is not a ValidationError but an actual Error.

Reply:
Alternative could you not check if status === 200 return the json, else return a error?
Seems more failsafe in this scenario.

Reply:
> This is not a ValidationError but an actual Error.

I agree, I was not able to find a similar error type we use for GraphQL that should represent Internal Server Error. Are you familiar with any?

Reply:
> Alternative could you not check if status === 200 return the json, else return a error?

Good point. We can opt for this for easier checking. Thank you for the suggestion.

Reply:
> There is a res.ok, it would be best to use that.

Will it be still true on `res.code === 204`?

Reply:
Yes, it has `200-299` range https://developer.mozilla.org/en-US/docs/Web/API/Response/ok

Reply:
Oh, then I think we need to be specific with 200 since 204 is an empty response.

Reply:
Must have missed something, why would it return `204`?

Reply:
Not really with the ins and outs of the API we are using but as a failsafe option.

Reply:
Are we not using API provided by the platform team, there should be defined valid responses?

Reply:
We don't have yet documentation for the relevant API. We don't even have the exact 400-500 range error codes it might produce but to at least handle the situations we provided the conditions.

Reply:
If we don't know exact error codes we should return the link or throw 500 and log internally what happened. What if we start getting `401` and `403` from that API, those are not validation errors (as returned now) but auth errors. It just seems wrong to be specific in some cases and then not so specific in others when we don't actually know anything.

Reply:
Don't want to hold the discussion on technicality, maybe something to think about in the future.

Reply:
Totally agree with that, we need to come up with something on how we should proceed for such cases. Thanks for the thoughts!